gax-java/gax/src/main/java/com/google/api/gax/rpc/ApiClientHeaderProvider.java

```diff
@@ -68,6 +70,9 @@ protected ApiClientHeaderProvider(Builder builder) {
       headersBuilder.put(QUOTA_PROJECT_ID_HEADER_KEY, builder.getQuotaProjectIdToken());
     }
 
+    if (builder.getApiVersionToken() != null) {
```

Comment:
I think it should be `Token != null && token != ""`

Reply:
I am not exactly sure I am following your thoughts here. 
If you are talking about the public setter in the Builder, then I think we are safe here. This constructor is protected and is meant to enforce construction through the builder.

or is this what you have in mind?
```
ApiClientHeaderProvider emptyProvider = ApiClientHeaderProvider.newBuilder().setApiVersionToken("")
        .build();
```
I don't think it's this HeaderProvider's responsibility to make any judgement on the header values. This logic belongs to `AbstractServiceStubSettingsClassComposer.java` via `service.hasApiVersion()`. I'll add in relevant tests to demonstrate.

Reply:
I am not exactly sure I am following your thoughts here. 
If you are talking about the public setter in the Builder, then I think we are safe here. This constructor is protected and is meant to enforce construction through the builder.

or is this what you have in mind?
```
ApiClientHeaderProvider emptyProvider = ApiClientHeaderProvider.newBuilder().setApiVersionToken("")
        .build();
```
I don't think it's this HeaderProvider's responsibility to make any judgement on the header values. This logic belongs to `AbstractServiceStubSettingsClassComposer.java` via `service.hasApiVersion()`. I'll add in relevant tests to demonstrate.

Reply:
> I don't think it's this HeaderProvider's responsibility to make any judgement on the header values. This logic belongs to AbstractServiceStubSettingsClassComposer.java via service.hasApiVersion(). I'll add in relevant tests to demonstrate.

The composer only checks the api versionvalue from the proto file. This case is for any runtime configurations that the user adds to the client library. We currently treat any service configuration where api_version of `""` as null and I am currently thinking that it would make sense for the same behavior here.

IIUC, ClientContext does make judgement on the header values and does some validation: https://github.com/googleapis/sdk-platform-java/blob/fe0481ec599f26f6cfe4df48b404da20ee447f57/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L297-L308. I was thinking that it should be enforced, but that isn't required from this PR.